vpr/src/place/place_delay_model.h

```diff
@@ -75,19 +72,20 @@ class OverrideDelayModel : public PlaceDelayModel {
     float delay(int from_x, int from_y, int from_pin, int to_x, int to_y, int to_pin) const override;
     void dump_echo(std::string filepath) const override;
 
-    void read(const std::string& /*file*/) override {
-        VPR_THROW(VPR_ERROR_ROUTE, "OverrideDelayModel::read unimplemented");
-    }
-    void write(const std::string& /*file*/) const override {
-        VPR_THROW(VPR_ERROR_ROUTE, "OverrideDelayModel::write unimplemented");
-    }
+    void read(const std::string& file) override;
+    void write(const std::string& file) const override;
 
   public: //Mutators
+    void set_base_delay_model(std::unique_ptr<DeltaDelayModel> base_delay_model);
+    const DeltaDelayModel* base_delay_model() const;
+    float get_delay_override(int from_type, int from_class, int to_type, int to_class, int delta_x, int delta_y) const;
+    void set_delay_override(int from_type, int from_class, int to_type, int to_class, int delta_x, int delta_y, float delay);
+
   private:
-    std::unique_ptr<PlaceDelayModel> base_delay_model_;
+    std::unique_ptr<DeltaDelayModel> base_delay_model_;
```

Comment:
Is it necessary to change this from the abstract PlaceDelayModel to DeltaDelayModel? The idea was that any (potentially non-delta delay model) could be combined with this class to provide fix-up overrides.

Reply:
Serialization would require a mirroring of the inner model type.  We could create a string registry of types, and then serialize the string.  However, I didn't want to take that additional complexity right now given how few models we have.

Do we want to add such a registry now?